client/services/utils/local_settings.js

```diff
@@ -0,0 +1,73 @@
+import EventEmitter from 'event-emitter-es6'
+
+const STORAGE_KEY = 'localsettings'
+const DATA_TYPE_KEY = 'type'
+const VALUE_KEY = 'value'
+const NAME_KEY = 'name'
+class LocalSettings extends EventEmitter {
+  constructor ($window, $rootScope) {
+    super()
+    Object.assign(this, { $window, $rootScope })
+    const settingsFromSession = this.getFromLocalStorage()
+    if (settingsFromSession !== undefined) {
+      this.settings = settingsFromSession
+    } else {
+      this.settings = { }
+    }
+  }
+
+  // example:
+  /**
+   * example input: settingKey: 'scoresheel-autoscroll' settingValueTypePair: {value: true, type: 'boolean'}
+   * @param {string} settingsKey
+   * @param {{value:*,type:string}} settingValueTypePair
+   */
+  update (settingsKey, settingValueTypePair) {
+    this.settings[settingsKey] = settingValueTypePair
+    this.emit(`${settingsKey}`)
+    this._saveToLocalStorage()
+  }
+  /**
+   * saves this service's settings object to the session storage
+   */
+  _saveToLocalStorage () {
+    this.$window.sessionStorage.setItem(STORAGE_KEY, JSON.stringify(this.settings))
+  }
+
+  /**
+   * for example: 'scoresheet-autoscroll'
+   * @param {string} settingsKey
+   */
+  get (settingsKey) {
+    if (this.settings.hasOwnProperty(settingsKey)) {
+      return this.settings[settingsKey]
+    } else {
+      return this.getFromLocalStorage(settingsKey)
```

Comment:
You should save this to `this.settings[settingsKey]`, for the next time it will be requested.